hudi-client/hudi-client-common/src/main/java/org/apache/hudi/io/HoodieAppendHandle.java

```diff
@@ -461,11 +464,11 @@ protected void appendDataAndDeleteBlocks(Map<HeaderMetadataType, String> header,
             ? HoodieRecord.RECORD_KEY_METADATA_FIELD
             : hoodieTable.getMetaClient().getTableConfig().getRecordKeyFieldProp();
 
-        blocks.add(getBlock(config, pickLogDataBlockFormat(), recordList, getUpdatedHeader(header, blockSequenceNumber++, taskContextSupplier.getAttemptIdSupplier().get()), keyField));
+        blocks.add(getBlock(config, pickLogDataBlockFormat(), recordList, getUpdatedHeader(header, blockSequenceNumber++, attemmptNo), keyField));
```

Comment:
I guess incrementing `attemptNo` is taken care of by the engone context itself?

Reply:
yes, it is. I tested it e2e. its hard to simulate in function tests though :(